src/careamics/dataset_ng/patch_extractor/image_stack/czi_image_stack.py

```diff
@@ -0,0 +1,297 @@
+import re
+from collections.abc import Sequence
+from pathlib import Path
+from typing import Any, Literal
+
+import numpy as np
+from numpy.typing import NDArray
+from pylibCZIrw.czi import CziReader, Rectangle, open_czi
+
+
+class CziImageStack:
+    """
+    A class for extracting patches from an image stack that is stored as a CZI file.
+
+    Parameters
+    ----------
+    data_path : str or Path
+        Path to the CZI file.
+
+    scene : int, optional
+        Index of the scene to extract. If not specified, the entire image covering
+        all scenes present in the CZI file will be extracted.
+        The scene can also be provided as part of `data_path` by appending an `"@"`
+        followed by the scene index to the filename.
+
+    depth_axis : {"none", "Z", "T", "auto"}, default: "auto"
+        Which axis to use as depth-axis for providing 3-D patches.
+
+        - `"none"`: Only provide 2-D patches. If a Z or T dimension is present in the
+          data, they will be combined into the sample dimension `S`.
+        - `"Z"`: Use the Z-axis as depth-axis. If a T axis is present as well, it will
+          be used as sample dimensions `S`.
+        - `"T"`: Use the T-axis as depth-axis. If a Z axis is present as well, it will
+          be used as sample dimensions `S`.
+        - `"auto"`: Automatically uses a Z or T axis present in the data as depth axis.
+          If both are present, the Z axis takes precedence.
+
+    Attributes
+    ----------
+    source : Path
+        Path to the CZI file, including the scene index if specified.
+    data_path : Path
+        Path to the CZI file without scene index.
+    scene : int or None
+        Index of the scene to extract, or None if not specified.
+    data_shape : Sequence[int]
+        The shape of the data in the order `(SC(Z)YX)`.
+    axes : str
+        The axes in the CZI file corresponding to the dimensions in `data_shape`.
+        Possible axes are `C`, `T`, `Z`, `Y`, `X`, and `S`. The axis `S` (sample)
```

Comment:
Can "C" be absent? The example in the configuration factory stated otherwise.

Reply:
It cannot. I added a list of all possible combinations.